src/nvim/screen.c

```diff
@@ -4374,6 +4374,11 @@ static void grid_put_linebuf(ScreenGrid *grid, int row, int coloff, int endcol,
 
   screen_adjust_grid(&grid, &row, &coloff);
 
+  // Safety check. Avoids clang warnings down the call stack.
+  if (grid->chars == NULL || row >= grid->Rows || col >= grid->Columns) {
+    return;
```

Comment:
should we leave a hint in the log?

    DLOG("invalid state, skipped");

Reply:
fixed